middleware.ts

```diff
@@ -0,0 +1,58 @@
+import { NextResponse } from 'next/server';
+
+import type { NextRequest } from 'next/server';
+
+export function middleware(request: NextRequest) {
+  const { pathname } = request.nextUrl;
+
+  switch (pathname) {
+    case '/api/elections':
+      return _checkElectionParameter(request);
+    case '/api/elections/page':
+      return _checkElectionPageParameter(request);
+    default:
+      NextResponse.next();
+  }
+}
+
+function _checkElectionParameter(
+  request: NextRequest,
+): NextResponse | undefined {
+  if (request.method !== 'GET') {
+    methodNotAllowed();
```

Comment:
The methodNotAllowed is not being sent when it happens, to fix it just add a return here, like:

```suggestion
    return methodNotAllowed();
```